File: src/Constants/winmasks.go
package Constants

import (
	"github.com/utkarsh-naman/3T/src/model"
)

const winRow1Mask model.State = 0b111000000
const winRow2Mask model.State = 0b000111000
const winRow3Mask model.State = 0b000000111
const winCol1Mask model.State = 0b100100100
const winCol2Mask model.State = 0b010010010
const winCol3Mask model.State = 0b001001001
const winDia1Mask model.State = 0b100010001
const winDia2Mask model.State = 0b001010100

var WinMasks = [8]model.State{winRow1Mask, winRow2Mask, winRow3Mask, winCol1Mask, winCol2Mask, winCol3Mask, winDia1Mask, winDia2Mask}

File: New folder/src/constants/win_masks.rs
use crate::model::state_model::State;

pub static WIN_ROW1_MASK: State = 0b111000000;
pub static WIN_ROW2_MASK: State = 0b000111000;
pub static WIN_ROW3_MASK: State = 0b000000111;
pub static WIN_COL1_MASK: State = 0b100100100;
pub static WIN_COL2_MASK: State = 0b010010010;
pub static WIN_COL3_MASK: State = 0b001001001;
pub static WIN_DIAG1_MASK: State = 0b100010001;
pub static WIN_DIAG2_MASK: State = 0b001010100;

pub static WIN_MASKS: [State; 8] = [WIN_ROW1_MASK, WIN_ROW2_MASK, WIN_ROW3_MASK, WIN_COL1_MASK, WIN_COL2_MASK, WIN_COL3_MASK, WIN_DIAG1_MASK, WIN_DIAG2_MASK];